posts/zzz_DO_NOT_EDIT_roxy.shinyl.../roxy.shinylive_-_shinylive_applications_in_roxygen_documentation.qmd

```diff
@@ -0,0 +1,80 @@
+
+![](roxy.shinylive.gif)
+
+Continuing my exploration of WebR (see my previous post [here](https://pharmaverse.github.io/blog/posts/2024-05-08_tlg_catalog_webr/tlg_catalog_webr.html)), I'm happy to introduce a new tool for Shiny package developers - [`roxy.shinylive`](https://insightsengineering.github.io/roxy.shinylive/). The package is designed for anyone building their Shiny applications or modules as an R package. With just a few lines of code, you can easily embed an `iframe` to a Shinylive application based on the code from the "Examples" section of your documentation.
+
+Typically, you might have something like this:
+```{r}
+#| eval: false
+#' This is a super app constructor.
+#' @param ... something
+#' @return shiny app object
+#' @examples
+#' if (interactive()) {
+#'   my_super_app()
+#' }
+my_super_app <- function(...) {
+  ...
+  shiny::shinyApp(...)
+}
+```
+
+Now, let's use `@examplesShinyLive` tag and make small adjustments to the example code to make it work in Shinylive:
+```{r}
+#| eval: false
+#' @examplesShinyLive
+#' library(mypackage)
+#' interactive <- function() TRUE
+#' {{ next_example }}
+#' @examples
+#' (...)
+```
+
+Voilà! Now your documentation includes Shinylive app! As a result, the end-users are able to see your application in action without needing to install anything. This makes your package more accessible and closer to the end-users.
+
+In addition, the package exports `create_shinylive_url()` function, which creates an URL based on the application code as a string. This opens up more possibilities, such as embedding application(s) in README files, vignettes or even outside of package documentation. Combined with `knitr::knit_code$get("<chunk id>")` and `knitr::include_url()`, you can reuse other chunk(s) code to embed `iframe`s in RMarkdown or Quarto documents.
+
+For a practical example of implementation, please see the documentation of [`teal.modules.general`](https://insightsengineering.github.io/teal.modules.general/main/) or [`teal.modules.clinical`](https://insightsengineering.github.io/teal.modules.clinical/main/) packages with more to come.
+
+Special thanks to Sam Parmar from Pfizer - the author of the [`lzstring`](https://parmsam.github.io/lzstring-r/) package, which makes encoding / decoding possible. Yet another example of cross-pharma collaboration!
+
+PS. Yes - it's coming to CRAN soon.
+
+PS2. I've been also thinking about a similar solution for non-Shiny R codes. See [this](https://github.com/r-lib/pkgdown/issues/2348) and [this](https://github.com/r-wasm/webr/issues/356) issues for more updates.
+
+I’d love for you to try out `roxy.shinylive` and see how it can enhance your Shiny package development workflow. Whether you have feedback, suggestions, or feature requests, please feel free to file an issue on the https://github.com/insightsengineering/roxy.shinylive, or join our [`pharmaverse` Slack channel](https://pharmaverse.slack.com/) to stay up to date and be part of the discussion.
```

Comment:
It's really cool! 
What would the placeholder for such an iframe be in the pdf version of the manual?
What does it look like when I look at the help page from the r console in rstudio, i.e. using `?function`

Reply:
For pdf it would be just an empty space. It's in the docs [here](https://insightsengineering.github.io/roxy.shinylive/main/reference/tag-examplesShinylive.html#:~:text=and%20for%20HTML%20outputs%3A%20an%20iframe%20with%20the%20application). If you are curious about implementation -> [here](https://github.com/insightsengineering/roxy.shinylive/blob/c1f1a4eae43f61f52f549a0705f5ab6936b4b782/R/tag_examplesShinylive.R#L218-L219)

In the R console you can do the following:
`? roxy.shinylive::create_shinylive_url` -> it would open a manual of `create_shinylive_url()` which is also [here](https://insightsengineering.github.io/roxy.shinylive/main/reference/create_shinylive_url.html)
One can also do `help("tag-examplesShinylive")` to see [this](https://insightsengineering.github.io/roxy.shinylive/main/reference/tag-examplesShinylive.html). I don't know how to open the latter via `?` - that's why I encouraged the use of pkgdown help throughout the article.

I think I don't want to mention that here not to make it too detailed or become a technical writing. That's just to pitch an idea if you know what I mean